chainer/dataset/convert.py

```diff
@@ -39,12 +39,15 @@ def to_device(device, x):
 def concat_examples(batch, device=None, padding=None):
     """Concatenates a list of examples into array(s).
 
-    Dataset iterator yields a list of examples. If each example is an array,
-    this function concatenates them along the newly-inserted first axis (called
-    `batch dimension`) into one array. The basic behavior is same for examples
-    consisting of multiple arrays, i.e., corresponding arrays of all examples
-    are concatenated.
-
+    This function converts "an arrays of tuples" into "a tuple of arrays".
+    Given an list of examples each of which consists of
```

Comment:
Given a list of examples, each of which consists of
a list of elements, this function makes an array
by taking the element in the same position from each example
and concatenates them along the newly-inserted first axis
(called `batch dimension`) into one array.
It repeats this for all positions and returns the resulting arrays.